Update 2025-07-14-kuzu-v-0.11.0.md (#130)
kuzudb/kuzu-blog: src/content/post/2025-07-14-kuzu-v-0.11.0.md

## Vector index and FTS improvements

> This release includes extensive improvements to both the vector and full-text search (FTS) indeces. We list some of the highlights below.

Reviewer: ```suggestion
This release includes extensive improvements to both the vector and full-text search (FTS) indices. We list some of the highlights below.
```

in that table will automatically trigger an update in the associated index.

If your workflows involve manually dropping and recreating indices on tables that are updated
with new data, you can now just insert the data, and Kuzu will handle the rest.

### Filtered vector search with arbitrary Cypher queries

> Earlier versions of vector search only supported simple, single-table filters. In v0.11.0, filters can now be expressed as arbitrary Cypher queries. Consider the following

Reviewer: ```suggestion
Earlier versions of vector search only supported simple predicate-based filters on a given table. In v0.11.0, we generalize this so that filters can be expressed as arbitrary Cypher queries. Consider the following
```

This feature should make it simpler to update your database with new embeddings as you obtain new data,
without having to write custom Python scripts to manage your embedding pipeline outside of Kuzu. To
learn more about which model providers are supported, see the [docs](https://dev-docs.kuzudb.com/extensions/llm/).

## Azure 

We've added initial support for scanning data from Azure Blog Storage and Azure Data Lake Storage (ADLS).
If you're using Kuzu in Azure environments, check out this feature in the [docs](https://dev-docs.kuzudb.com/extensions/azure/) and reach out to us via a [GitHub issue](https://github.com/kuzudb/kuzu/issues) or discussion if you have any feedback.

## Swift API

Kuzu is designed to run anywhere. With this goal in mind, we're excited to introduce
our new Swift API, allowing you to embed and run Kuzu on Apple devices.

## Other Cypher features

### Create table as

> Similar to PostgreSQL, Kuzu now support creating and populating tables through `CREATE NODE/REL TABLE AS <subquery>`. The created table have the same column names and types as the result of `subquery`. 

Reviewer: ```suggestion
Similar to PostgreSQL, Kuzu now support creating and populating tables through `CREATE NODE/REL TABLE AS <subquery>`. The created table has the same column names and types as the result of `subquery`. 
```

> For example, the following query creates a new node table `YoungPerson` with a subset of record from `Person` table. 

Reviewer: ```suggestion
For example, the following query creates a new node table `YoungPerson` with a subset of records from the `Person` table. 
```

### Alter relationship table add/drop connection

> You can now modify the connections (from/to) of a relationship table schema after creation. This enables more flexible graph evolution.

Reviewer: ```suggestion
You can now modify the connections (from/to) of a relationship table schema after creation. This enables more flexible evolution of your graph schema.
```
